sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -261,6 +261,20 @@ trait ExternalCatalog {
       table: String,
       partialSpec: Option[TablePartitionSpec] = None): Seq[CatalogTablePartition]
 
+  /**
+   * List the metadata of all partitions that belong to the specified table
+   * and the giving part names,assuming it exists.
+   *
+   * @param db database name
+   * @param table table name
+   * @param partitionNames partNames e.g.  ('a=1/b=1', 'a=1/b=2')
```

Comment:
1. Remove extra space after `e.g.`
2. Why `partitionNames` (with `Names` suffix` while `db` and `table`?  Ah, it's `partNames` actually. Why not `parts` or even `partitions`?